### PR TITLE
fix(styles): add focus color on datepicker component

### PR DESCRIPTION
**Pull request by davidritter-dotcom**

### CODE REVIEW

File: packages/styles/src/components/datepicker.scss [hunk 1]
```diff
@@ -98,6 +98,9 @@ span.ngb-dp-navigation-chevron {
   width: 40px !important;
   height: 40px !important;
   line-height: 40px;
+  &:focus-visible{
+    outline: forms.$input-focus-color solid forms.$input-focus-outline-thickness;
+  }
```
Comment (imagoiq) [inline packages/styles/src/components/datepicker.scss:103]: With keyboard navigation, it doesn't work (check Firefox as the default color is blue, it's more obvious).

**Review by imagoiq — CHANGES_REQUESTED** [review]
Also, don't forget to lint. I would advise for doing it automatically with your IDE to avoid forgetting about it.

File: packages/styles/src/components/datepicker.scss [hunk 2]
```diff
@@ -139,6 +146,9 @@ span.ngb-dp-navigation-chevron {
     height: datepicker.$ngb-dp-icon-size;
     transform: none;
   }
+  &:focus-visible {
+    outline: forms.$input-focus-outline-thickness solid var(--post-contrast-color);
+  }
```
Comment (imagoiq) [inline packages/styles/src/components/datepicker.scss:151]: This one is not compliant with WCAG. Check this article: https://www.sarasoueidan.com/blog/focus-indicators/

You need the same style as buttons.

**Review by imagoiq — CHANGES_REQUESTED** [review]
Need another style for the trigger button.

**Comment by gfellerph** [comment]
Wait for [#1982](https://github.com/swisspost/design-system/issues/1982)

The style should be updated according to https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=16545%3A53350&mode=design&t=loOePihAoe8Rf8Bz-1

**Review by gfellerph — CHANGES_REQUESTED** [review]
Wait for https://github.com/swisspost/design-system/issues/1982

The style should be updated according to https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=16545%3A53350&mode=design&t=loOePihAoe8Rf8Bz-1

**Comment by gfellerph** [comment]
@davidritter-dotcom implement with the current definition in Figma.

**Comment by gfellerph** [comment]
@davidritter-dotcom design is ready. This is no longer blocked.

**Review by gfellerph — CHANGES_REQUESTED** [review]
Fix linting errors and update according to current Design. If that's done, please re-request review.

**Comment by davidritter-dotcom** [comment]
Waiting for #2855 because of the change in the foucs-style mixin.

**Review by gfellerph — CHANGES_REQUESTED** [review]
There some points I noticed that don't seem to be working right now
- Focus ring should not be visible when clicking with a mouse, 
- Focus styles are not visible if a date is selected (select a date and open the picker again, the focus ring should be visible when interacting with keyboard)
- When the picker is open, the date field (and in the demo some more components) get a focus ring as well image. 
  This should not be the case

**Comment by oliverschuerch** [comment]
@davidritter-dotcom: This task is in progress quite long now. Do you need help with something?
Do you need preciser instructions from one of the reviewer (@imagoiq and @gfellerph)?

Let me know if we can help.

**Comment by davidritter-dotcom** [comment]
The focus on the select itself gets fixed when a decision is reached on how to apply focus styles to form-select and form-control.

**Review by gfellerph — APPROVED** [review]

**Pull request merged**